### PR TITLE
Release 2.3.1

### LINKED ISSUE #57: Prepare a new release
Decide if 2.3.1 or 2.4.0.  Taking a quick glance at the commits, it was mostly release tooling and helm chart fixes that went in.  cc @cofyc if there was something major I missed.

/assign

### PR DESCRIPTION
**Pull request by cofyc**
fixes #57 

- update README.md and CHANGELOG.md
- upgrade helm chart to 2.3.1

### CODE REVIEW

**Comment by cofyc** [comment]
/assign @msau42 
By the way, after this PR is merged, could you create a tag `2.3.1` and push it?
Our [release job](https://testgrid.k8s.io/sig-storage-local-static-provisioner#release-build) should build an image for the new tag automatically. If it fails I will build the image manually.

File: README.md [hunk 1]
```diff
@@ -78,17 +78,19 @@ See [Best Practices](docs/best-practices.md).
...
 
-| Provisioner version | K8s version   | Reason                    |
-| ------------------- | ------------- | ------------------------- |
-| [2.3.0][4]          | 1.12          | fs on block support       |
-| [2.2.0][3]          | 1.10          | Beta API default, block   |
-| [2.0.0][2]          | 1.8, 1.9      | Mount propagation         |
-| [1.0.1][1]          | 1.7           |                           |
+| Provisioner version | K8s version   | Reason                      |
+| ------------------- | ------------- | --------------------------- |
+| [2.3.1][5]          | 1.14          | local persistent volumes GA |
```
Comment (msau42) [inline README.md]: There's no new features in 2.3.1 that depend on 1.14 is there?  Should we just update 2.3.0 to 2.3.1?

Comment (cofyc) [reply README.md]: fixed, you are right, it is version compatibility

File: README.md [hunk 2]
```diff
@@ -80,6 +80,7 @@ Recommended provisioner versions with Kubernetes versions
 
 | Provisioner version | K8s version   | Reason                    |
 | ------------------- | ------------- | ------------------------- |
+| [2.3.1][5]          | 1.12          |                           |
 | [2.3.0][4]          | 1.12          | fs on block support       |
```
Comment (msau42) [inline README.md]: Maybe just replace 2.3.0 with 2.3.1? It's the same minor release train.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
 # HEAD
 
+# [v2.3.1](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/releases/tag/v2.3.1)
+
+Image updates:
+- Fix an issue in block devices cleanup by Kubernetes Job
+  ([#60](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/pull/60))
+
+Deployment updates:
+- Support pod security policy
```
Comment (msau42) [inline CHANGELOG.md]: Do any of these deployment updates depend on newer than 1.12?

File: README.md (hunk 2 above)
Comment (cofyc) [reply README.md]: done

File: CHANGELOG.md (hunk 3 above)
Comment (cofyc) [reply CHANGELOG.md]: no

**Pull request merged**